model/load_balancer.rb

```diff
@@ -21,14 +22,24 @@ def hyper_tag_name(project)
   def add_vm(vm)
     DB.transaction do
       incr_update_load_balancer
+      incr_rewrite_dns_records
       super
     end
   end
 
   def detach_vm(vm)
     DB.transaction do
-      DB[:load_balancers_vms].where(load_balancer_id: id, vm_id: vm.id).delete(force: true)
+      DB[:load_balancers_vms].where(load_balancer_id: id, vm_id: vm.id).update(state: "evacuating")
```

Comment:
Ah I see the delete just doesn't matter because it becomes update, ignore my previous comment I suppose; I don't think the intermediate state between the two patches being ever so slightly suboptimal is worth a change.